RecoTracker/LSTCore/src/alpaka/Event.dev.cc

```diff
@@ -735,32 +735,39 @@ void Event::createPixelTriplets() {
 
   // TODO: check if a map/reduction to just eligible pLSs would speed up the kernel
   // the current selection still leaves a significant fraction of unmatchable pLSs
-  for (unsigned int i = 0; i < nInnerSegments; i++) {  // loop over # pLS
-    int8_t pixelType = pixelTypes[i];                  // Get pixel type for this pLS
-    int superbin = superbins[i];                       // Get superbin for this pixel
-    if ((superbin < 0) or (superbin >= (int)size_superbins) or (pixelType > 2) or (pixelType < 0)) {
+  for (unsigned int i = 0; i < nInnerSegments; i++) {                           // loop over # pLS
+    ::lst::PixelType pixelType = static_cast<::lst::PixelType>(pixelTypes[i]);  // Get pixel type for this pLS
+    int superbin = superbins[i];                                                // Get superbin for this pixel
+    if ((superbin < 0) or (superbin >= (int)size_superbins) or
+        ((pixelType != ::lst::PixelType::kHighPt) and (pixelType != ::lst::PixelType::kLowPtPosCurv) and
+         (pixelType != ::lst::PixelType::kLowPtNegCurv))) {
       connectedPixelSize_host[i] = 0;
       connectedPixelIndex_host[i] = 0;
       continue;
     }
 
     // Used pixel type to select correct size-index arrays
-    if (pixelType == 0) {
-      connectedPixelSize_host[i] =
-          pixelMapping_.connectedPixelsSizes[superbin];  // number of connected modules to this pixel
-      auto connectedIdxBase = pixelMapping_.connectedPixelsIndex[superbin];
-      connectedPixelIndex_host[i] =
-          connectedIdxBase;  // index to get start of connected modules for this superbin in map
-    } else if (pixelType == 1) {
-      connectedPixelSize_host[i] =
-          pixelMapping_.connectedPixelsSizesPos[superbin];  // number of pixel connected modules
-      auto connectedIdxBase = pixelMapping_.connectedPixelsIndexPos[superbin] + pixelIndexOffsetPos;
-      connectedPixelIndex_host[i] = connectedIdxBase;  // index to get start of connected pixel modules
-    } else if (pixelType == 2) {
-      connectedPixelSize_host[i] =
-          pixelMapping_.connectedPixelsSizesNeg[superbin];  // number of pixel connected modules
-      auto connectedIdxBase = pixelMapping_.connectedPixelsIndexNeg[superbin] + pixelIndexOffsetNeg;
-      connectedPixelIndex_host[i] = connectedIdxBase;  // index to get start of connected pixel modules
+    unsigned int connectedIdxBase;
+    switch (pixelType) {
+      case ::lst::PixelType::kHighPt:
+        connectedPixelSize_host[i] =
+            pixelMapping_.connectedPixelsSizes[superbin];  // number of connected modules to this pixel
+        connectedIdxBase = pixelMapping_.connectedPixelsIndex[superbin];
+        connectedPixelIndex_host[i] =
+            connectedIdxBase;  // index to get start of connected modules for this superbin in map
```

Comment:
```suggestion
        // number of connected modules to this pixel
        connectedPixelSize_host[i] = pixelMapping_.connectedPixelsSizes[superbin];
        // index to get start of connected modules for this superbin in map
        connectedPixelIndex_host[i] = pixelMapping_.connectedPixelsIndex[superbin];
```
does away with not really necessary `connectedIdxBase`